File: CousinPCMS.Domain/SkusModel.cs
using Newtonsoft.Json;

namespace CousinPCMS.Domain
{
    public class SkusModel
    {
    }

    public class AttributeFilterRequest
    {
        public List<string> attributeNames { get; set; }
    }
    public class RelatedSkusModel
    {
        [JsonProperty("@odata.etag")]
        public string ODataEtag { get; set; }

        [JsonProperty("itemNo")]
        public string ItemNo { get; set; }

        [JsonProperty("relatedItemNo")]
        public string RelatedItemNo { get; set; }

        [JsonProperty("relationType")]
        public string RelationType { get; set; }

        [JsonProperty("relatedSKUName")]
        public string RelatedSKUName { get; set; }

        [JsonProperty("itemManufactureRef")]
        public string ItemManufactureRef { get; set; }

        [JsonProperty("itemObsolte")]
        public bool ItemObsolte { get; set; }

        [JsonProperty("itemIsUnavailable")]
        public bool ItemIsUnavailable { get; set; }

        [JsonProperty("isrelatedSKU")]
        public bool IsrelatedSKU { get; set; }

        public int skuID { get; set; }
    }

    public class DeleteSkusRequestModel
    {
        public string itemno { get; set; }
    }

    public class UpdateItemRelatedSkuModel
    {
        public string itemNo { get; set; }
        public string relatedItemNo { get; set; }
        public bool isrelatedSKU { get; set; }
    }

    public class UpdateSkuItemRequestModel
    {
        public string akiitemid { get; set; }
        public string akiProductID { get; set; }
        public string akiCategoryID { get; set; }
        public string akiManufacturerRef { get; set; }
        public int akiListOrder { get; set; }
        public bool akiObsolete { get; set; }
        public bool akiWebActive { get; set; }
        public string akiImageURL { get; set; }
        public string akiCommodityCode { get; set; }
        public int akiGuidePriceTBC { get; set; }
        public int akiGuideWeightTBC { get; set; }
        public bool akicurrentlyPartRestricted { get; set; }
        public string akiCountryofOrigin { get; set; }
        public bool akiPrintLayoutTemp { get; set; }
        public string akiAlternativeTitle { get; set; }
        public string akiCompetitors { get; set; }
        public bool akiPriceBreaks { get; set; }
        public int akiItemShippingWeight { get; set; }
        public int akiItemPriceSiteSellPrice { get; set; }
        public string akiTemplateID { get; set; }
        public string akiSKUDescription { get; set; }
        public string akiPricingFormula { get; set; }
        public string akiPriceBreak { get; set; }
        public string akiPriceGroup { get; set; }
        public bool akiSKUIsActive { get; set; }
    }

    public class SkusRelationTypeModel
    {
        [JsonProperty("@odata.etag")]
        public string ODataEtag { get; set; }

        [JsonProperty("relationCode")]
        public string RelationCode { get; set; }

        [JsonProperty("relationDescription")]
        public string RelationDescription { get; set; }
    }
    public class SkusLayoutModel
    {
        [JsonProperty("@odata.etag")]
        public string ODataEtag { get; set; }

        [JsonProperty("itemId")]
        public int ItemId { get; set; }

        [JsonProperty("templateCode")]
        public string TemplateCode { get; set; }

        [JsonProperty("layoutDescription")]
        public string LayoutDescription { get; set; }
    }

    public class SkusLinkedURlModel
    {
        [JsonProperty("@odata.etag")]
        public string odataetag { get; set; }
        public int skuitemURLid { get; set; }
        public string skuitemID { get; set; }
        public string linkText { get; set; }
        public string linkURL { get; set; }
        public string toolTip { get; set; }
        public string linkType { get; set; }
        public int listorder { get; set; }
    }

    public class SkusAdditionalImageModel
    {
        [JsonProperty("@odata.etag")]
        public string odataetag { get; set; }
        public int skuImageID { get; set; }

        [JsonProperty("skuitemID")]
        public string skuItemID { get; set; }

        public string imageName { get; set; }
        public string imageURL { get; set; }
        public int listorder { get; set; }
    }

    public class AddSkuAdditionalImageRequestModel
    {
        public string skuItemID { get; set; }
        public string imageURL { get; set; }
        public string imagename { get; set; }
        public int listorder { get; set; }
    }

    public class UpdateSkuAdditionalImageRequestModel
    {
        public int skuImageID { get; set; }
        public int oldlistorder { get; set; }
        public int newlistorder { get; set; }
    }

    public class UpdateSkuLinkUrlRequestModel
    {
        public int skuitemURLID { get; set; }
        public int oldlistorder { get; set; }
        public int newlistorder { get; set; }
    }

    public class AddSkuAdditionalLinkUrlRequestModel
    {
        public string skuItemID { get; set; }
        public string linkURL { get; set; }
        public string linkText { get; set; }
        public string toolTip { get; set; }
        public string linkType { get; set; }
        public int listorder { get; set; }
    }

    public class DeleteSkuURLRequestModel
    {
        public string skuItemID { get; set; }
        public string linkURL { get; set; }
    }

    public class DeleteSkuImageRequestModel
    {
        public string skuItemID { get; set; }
        public string imageURL { get; set; }
    }

    public class SkuAttributesModel
    {
        [JsonProperty("@odata.etag")]
        public string odataetag { get; set; }
        public int linkedId { get; set; }
        public string akiItemNo { get; set; }
        public string akiAttributeName { get; set; }
        public string akiAttributeValue { get; set; }
        public bool akiLink { get; set; }
    }

    public class AddSKULinkedAttributeRequestModel
    {
        public string akiItemNo { get; set; }
        public string akiAttributeName { get; set; }
        public string akiAttributeValue { get; set; }
        public bool akiLink { get; set; }
    }

    public class UpdateSKULinkedAttributeRequestModel
    {
        public int linkedid { get; set; }
        public string akiItemNo { get; set; }
        public string akiAttributeName { get; set; }
        public string akiAttributeValue { get; set; }
        public bool akiLink { get; set; }
    }

    public class UpdateSKUObsoleteRequestModel
    {
        public string akiitemid { get; set; }
        public bool akiObsolete { get; set; }
    }

    public class UpdateSKUUnavailableRequestModel
    {
        public string akiitemid { get; set; }
        public bool akiCurrentlyPartRestricted { get; set; }
    }

    public class DragDropSkuRequestModel
    {
        public string akiitemid { get; set; }
        public int oldlistorder { get; set; }
        public int newlistorder { get; set; }
        public string categoryid { get; set; }
        public int productid { get; set; }
    }

}

File: CousinPCMS.React/src/models/skusModel.ts
export interface SKuList {
  odataetag: string;
  akiitemid: string;
  skuName: string;
  priceUnitConversion: number;
  inventoryPostingGroup: string;
  shelfNo: string;
  itemDiscGroup: string;
  allowInvoiceDisc: boolean;
  statisticsGroup: number;
  commissionGroup: number;
  unitPrice: number;
  priceProfitCalculation: string;
  profit: number;
  unitCost: number;
  standardCost: number;
  lastDirectCost: number;
  indirectCost: number;
  costIsAdjusted: boolean;
  allowOnlineAdjustment: boolean;
  leadTimeCalculation: string;
  reorderPoint: number;
  maximumInventory: number;
  reorderQuantity: number;
  alternativeItemNo: string;
  unitListPrice: number;
  dutyDue: number;
  dutyCode: string;
  grossWeight: number;
  netWeight: number;
  unitsPerParcel: number;
  unitVolume: number;
  durability: string;
  freightType: string;
  tariffNo: string;
  dutyUnitConversion: number;
  countryRegionPurchasedCode: string;
  budgetQuantity: number;
  budgetedAmount: number;
  budgetProfit: number;
  blocked: boolean;
  blockReason: string;
  lastDateTimeModified: string;
  lastDateModified: string;
  lastTimeModified: string;
  priceIncludesVAT: boolean;
  vatBusPostingGrPrice: string;
  genProdPostingGroup: string;
  picture: string;
  countryRegionOfOriginCode: string;
  automaticExtTexts: boolean;
  noSeries: string;
  taxGroupCode: string;
  vatProdPostingGroup: string;
  reserve: string;
  globalDimension1Code: string;
  globalDimension2Code: string;
  stockoutWarning: string;
  preventNegativeInventory: string;
  variantMandatoryIfExists: string;
  applicationWkshUserID: string;
  coupledToCRM: boolean;
  gtin: string;
  defaultDeferralTemplateCode: string;
  lowLevelCode: number;
  lotSize: number;
  serialNos: string;
  lastUnitCostCalcDate: string;
  rolledUpMaterialCost: number;
  rolledUpCapacityCost: number;
  scrap: number;
  inventoryValueZero: boolean;
  discreteOrderQuantity: number;
  minimumOrderQuantity: number;
  maximumOrderQuantity: number;
  safetyStockQuantity: number;
  orderMultiple: number;
  safetyLeadTime: string;
  flushingMethod: string;
  replenishmentSystem: string;
  roundingPrecision: number;
  purchUnitOfMeasure: string;
  timeBucket: string;
  reorderingPolicy: string;
  includeInventory: boolean;
  manufacturingPolicy: string;
  reschedulingPeriod: string;
  lotAccumulationPeriod: string;
  dampenerPeriod: string;
  dampenerQuantity: number;
  overflowLevel: number;
  manufacturerCode: string;
  createdFromNonstockItem: boolean;
  purchasingCode: string;
  excludedFromCostAdjustment: boolean;
  serviceItemGroup: string;
  itemTrackingCode: string;
  lotNos: string;
  expirationCalculation: string;
  warehouseClassCode: string;
  specialEquipmentCode: string;
  putAwayTemplateCode: string;
  putAwayUnitOfMeasureCode: string;
  physInvtCountingPeriodCode: string;
  lastCountingPeriodUpdate: string;
  useCrossDocking: boolean;
  nextCountingStartDate: string;
  nextCountingEndDate: string;
  unitOfMeasureId: string;
  taxGroupId: string;
  salesBlocked: boolean;
  purchasingBlocked: boolean;
  itemCategoryId: string;
  inventoryPostingGroupId: string;
  genProdPostingGroupId: string;
  serviceBlocked: boolean;
  overReceiptCode: string;
  assemblyPolicy: string;
  routingNo: string;
  productionBOMNo: string;
  singleLevelMaterialCost: number;
  singleLevelCapacityCost: number;
  singleLevelSubcontrdCost: number;
  singleLevelCapOvhdCost: number;
  singleLevelMfgOvhdCost: number;
  overheadRate: number;
  rolledUpSubcontractedCost: number;
  rolledUpMfgOvhdCost: number;
  rolledUpCapOverheadCost: number;
  critical: boolean;
  commonItemNo: string;
  no2: string;
  searchDescription: string;
  description2: string;
  akiSKUID: number;
  akiProductID: string;
  akiCategoryID: string;
  akiSKUDescription: string;
  akiManufacturerRef: string;
  akiListOrder: number;
  akiObsolete: boolean;
  akiWebActive: boolean;
  akiImageURL: string;
  akiCommodityCode: string;
  akiGuidePriceTBC: number;
  akiGuideWeightTBC: number;
  akiSKUIsActive: boolean;
  akiCurrentlyPartRestricted: boolean;
  akiCountryOfOrigin: string;
  akiPrintLayoutTemp: boolean;
  akiAlternativeTitle: string;
  akiCompetitors: string;
  akiPriceBreaksTBC: boolean;
  itemclasscode: string;
  akiItemShippingWeight: number;
  itemUOfMeasureScheduleID: string;
  itemsellingUOfM: string;
  itemsupplierID: string;
  itemcreditoritemno: string;
  akiItemSiteID: string;
  akiItemPriceWebsiteSellPrice: number;
  itemstartingquantity: number;
  itemtype: string;
  itemvaluationMethod: string;
  akiItemPriceSiteSellPrice: number;
  akiItemShortDescription: string;
  akiTemplateID: number;
  akiAltSKUName: string;
  akiLayoutTemplate: string;
  akiPricingFormula: string;
  akiPriceBreak: string;
  akiPriceGroup: string;
  additionalImagesCount: number,
  urlLinksCount: number,
  akigpItemNumber: string,
}

export interface SkuRequestModel {
  skuName: string;
  akiSKUDescription?: string;
  akiManufacturerRef?: string;
  akiitemid?: string;
  akiProductID: string,
  akiCategoryID: string,
  akiListOrder?: string;
  akiObsolete: boolean;
  akiWebActive: boolean;
  akiCurrentlyPartRestricted: boolean;
  akiImageURL?: string;
  akiCommodityCode?: string;
  akiCountryOfOrigin?: string;
  akiSKUIsActive: boolean;
  akiGuidePriceTBC: number;
  akiGuideWeightTBC: number;
  akiAlternativeTitle?: string;
  akiLayoutTemplate?: string;
  akiCompetitors?: string;
  akiPriceBreak?: string;
  akiPriceGroup?: string;
  akiPricingFormula?: string;
  akiPrintLayoutTemp: boolean;
  additionalImagesCount?: string;
  urlLinksCount?: string;
  akiPriceBreaksTBC: boolean;
  akigpItemNumber: string,
}

export interface SkuListResponse {
  value: SKuList[];
  isSuccess: boolean;
  isError: boolean;
  exceptionInformation: any;
}

export interface LikedSkuModel {
  akiAttributeName: string;
  akiAttributeValue: string;
  akiItemNo: string;
  akiLink: boolean;
  odataetag?: string;
  linkedId?: number,
}


export interface RelatedSkuModel {
  akiitemid: string;
  akiObsolete?: boolean;
  akiCurrentlyPartRestricted?: boolean
}

export interface UpdateSKULinkedAttribute {
  linkedid?: number,
  akiItemNo: string,
  akiAttributeName: string,
  akiAttributeValue: string,
  akiLink: boolean
}

export interface RelatedSkuItem {
  relatedItemNo: string;
  relationType: string;
  relatedSKUName: string;
  itemManufactureRef: string;
  itemNo: string;
  itemObsolte: boolean;
  itemIsUnavailable: boolean;
  oDataEtag?: string;
  isrelatedSKU?: boolean;
}

export interface SKusRequestModelForProductOrderList {
  akiitemid?: string;
  oldlistorder: number,
  newlistorder: number
}


